test/common/access_log/access_log_formatter_test.cc

```diff
@@ -1572,6 +1600,62 @@ TEST(AccessLogFormatterTets, JsonFormatterTypedFilterStateTest) {
             fields.at("test_obj").struct_value().fields().at("inner_key").string_value());
 }
 
+TEST(AccessLogFormatterTest, FilterStateSpeciferTest) {
+  Http::TestRequestHeaderMapImpl request_headers;
+  Http::TestResponseHeaderMapImpl response_headers;
+  Http::TestResponseTrailerMapImpl response_trailers;
+  StreamInfo::MockStreamInfo stream_info;
+  stream_info.filter_state_->setData(
+      "test_key", std::make_unique<TestSerializedStringFilterState>("test_value"),
+      StreamInfo::FilterState::StateType::ReadOnly);
+  EXPECT_CALL(Const(stream_info), filterState()).Times(testing::AtLeast(1));
+
+  std::unordered_map<std::string, std::string> expected_json_map = {
+      {"test_key_plain", "test_value By PLAIN"},
+      {"test_key_typed", "\"test_value By TYPED\""},
+  };
+
+  std::unordered_map<std::string, std::string> key_mapping = {
+      {"test_key_plain", "%FILTER_STATE(test_key:PLAIN)%"},
+      {"test_key_typed", "%FILTER_STATE(test_key:TYPED)%"}};
+
+  JsonFormatterImpl formatter(key_mapping, false);
+
+  verifyJsonOutput(
+      formatter.format(request_headers, response_headers, response_trailers, stream_info),
+      expected_json_map);
+}
+
+TEST(AccessLogFormatterTest, TypedFilterStateSpeciferTest) {
```

Comment:
Nit: can you add one or two liner `//` comment above each `TEST` you are introducing with a short, intuitive explanation of what the test does?

Reply:
Friendly ping on this one.